src/main/java/bio/terra/service/dataset/flight/upgrade/predictableFileIds/ConvertToPredictableFileIdsUpdateFirestoreCollectionsStep.java

```diff
@@ -0,0 +1,50 @@
+package bio.terra.service.dataset.flight.upgrade.predictableFileIds;
+
+import bio.terra.service.dataset.Dataset;
+import bio.terra.service.dataset.DatasetService;
+import bio.terra.service.filedata.google.firestore.FireStoreDao;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+public class ConvertToPredictableFileIdsUpdateFirestoreCollectionsStep implements Step {
+
+  private final UUID datasetId;
+  private final DatasetService datasetService;
+  private final FireStoreDao fileDao;
+
+  public ConvertToPredictableFileIdsUpdateFirestoreCollectionsStep(
+      UUID datasetId, DatasetService datasetService, FireStoreDao fileDao) {
+    this.datasetId = datasetId;
+    this.datasetService = datasetService;
+    this.fileDao = fileDao;
+  }
```

Comment:
It looks like most `Step`s have all final fields. This means they could be defined using `record` instead of `class`. The main benefit is that you no longer need to write a constructor in addition to the field definitions, since record provides a constructor. In this case, you could write:
```suggestion
public record ConvertToPredictableFileIdsUpdateFirestoreCollectionsStep(UUID datasetId, DatasetService datasetService, FireStoreDao fileDao) implements Step {
```